lib/common.js

```diff
@@ -937,6 +937,46 @@ function getServerNicTags(cb) {
 	});
 }
 
+/*
+ * Query SAPI for the current update_channel. The callback function takes
+ * the form function(err, update_channel)
+ * This function can have a SAPI and bunyan log object passed as options,
+ * or will fallback to objects initialized by initSdcClients and self.log
+ * if those are not provided.
+ */
+function getSdcChannel(opts, cb) {
+	assert.object(opts, 'opts');
+	assert.optionalObject(opts.sapi, 'opts.sapi');
+	assert.optionalObject(opts.log, 'opts.log');
+	assert.func(cb, 'cb');
+
+	var sapi = opts.sapi;
+	var log = opts.log;
+	if (opts.sapi === undefined) {
+		sapi = this.SAPI;
+	}
+	if (opts.log === undefined) {
+		log = this.log;
+	}
+
+	var search = {};
+	search.name = 'sdc';
+	search.include_master = true;
+
+	sapi.listApplications(search, function (err, apps) {
+		if (err) {
+			log.error(err, 'failed to list applications');
+			return (cb(err));
+		}
+
+		log.debug('found sdc application');
+		assert.ok(apps.length <= 1);
+		assert.string(apps[0].metadata.update_channel);
```

Comment:
Oh, undocumented feature! Thanks - I'll use null, and check for that everywhere.

Reply:
This now looks like:
for manta-init:

```
[root@0cf080f3-51de-4103-a13b-e55a8853083d (coal-1:manta0) ~]# manta-init -e tim.foster@joyent.com
logs at /var/log/manta-init.log

and the logs say:

[2019-11-20T13:12:52.215Z]  INFO: manta-init/36382 on 0cf080f3-51de-4103-a13b-e55a8853083d: determining update_channel from sapi
[2019-11-20T13:12:52.316Z] DEBUG: manta-init/36382 on 0cf080f3-51de-4103-a13b-e55a8853083d: found sdc application
[2019-11-20T13:12:52.316Z]  INFO: manta-init/36382 on 0cf080f3-51de-4103-a13b-e55a8853083d: No channel data set in SAPI. Using default server update channel.
```

and for manta-adm when we ask for an image not on the default server channel:

```
[root@0cf080f3-51de-4103-a13b-e55a8853083d (coal-1:manta0) ~]# manta-adm update new.json
logs at /var/log/manta-adm.log
manta-adm: The following images were not found on the default updates channel and cannot be provisioned. (Use --skip-verify-channel to override): c53c9c3a-6e25-4fb1-ad4d-c64cf4c2ac30 (webapi)
[root@0cf080f3-51de-4103-a13b-e55a8853083d (coal-1:manta0) ~]# manta-adm update  --skip-verify-channel new.json
logs at /var/log/manta-adm.log
service "webapi"
  cn "564d4a02-d768-c09d-097e-4240ee50e1ae":
    reprovision zone 260119dc-d470-4eb1-8dea-1bec36749b6b
        (old image: ebfb30f0-debf-498d-8271-f2619921b706)
        (new image: c53c9c3a-6e25-4fb1-ad4d-c64cf4c2ac30)
Are you sure you want to proceed? (y/N):
```

Reply:
> Oh, undocumented feature! 

Yah, because existing SAPI databases before the introduction of `update_channel` of course didn't have a value set, and we wanted a way to test that state, but not really encourage operators to do so. At least I think that was the reasoning.